libs/x-content/src/test/java/org/elasticsearch/common/ParseFieldTests.java

```diff
@@ -71,4 +70,22 @@ public void testGetAllNamesIncludedDeprecated() {
         parseField = new ParseField("more_like_this", "mlt");
         assertThat(parseField.getAllNamesIncludedDeprecated(), arrayContainingInAnyOrder("more_like_this", "mlt"));
     }
+
+    public void testCompatibleField() {
+        ParseField field = new ParseField("new_name", "old_name")
+            .withRestApiCompatibilityVersions(RestApiCompatibleVersion.V_7);
+
+        assertTrue(field.match("new_name", LoggingDeprecationHandler.INSTANCE));
+        assertTrue(field.match("old_name", LoggingDeprecationHandler.INSTANCE));
+        assertWarnings("Deprecated field [old_name] used, expected [new_name] instead");
+
+        ParseField allDepField = new ParseField("dep", "old_name")
+            .withAllDeprecated()
+            .withRestApiCompatibilityVersions(RestApiCompatibleVersion.V_7);
```

Comment:
I assume this will need to be modified once `equalTo` and `onOrBefore` land ?  (can we test both cases once that lands)